### DISCUSSION TITLE
breaking changes 2.0

### DISCUSSION THREAD

**Discussion by mesqueeb** [vercel/turborepo#8307] [Help]
### Summary

looking at the release notes, it's not clear in 2.0 what the breaking changes are

**Comment by ehaynes99** [comment]
One is that it forces `packageManager` to be set in your `package.json`. Codemod doesn't allow semver, so you'll need to ensure everything in your CI pipeline uses the exact same version.

So e.g.

```diff
- npm i -g pnpm@^9
+ npm i -g pnpm@9.1.2
```
Or e.g. in github actions:
```diff
   steps:
     - uses: pnpm/action-setup@v3
       with:
-        version: 9
+        version: 9.1.2
```
This is a good practice anyway, but wasn't forced previously

**Comment by kylekz** [comment]
I noticed that env strict mode is now default. It seems that any `NEXT_PUBLIC_` keys I explicitly define in my environment vars get passed to Turbo and through to my Next app just fine, but any Vercel defaults like `NEXT_PUBLIC_VERCEL_ENV` do not, and I have to explicitly define this in my `turbo.json`. Is this intended behavior? I'm not disabling framework inference or anything, just ran the codemod to upgrade.

**Reply by trm217** [reply]
I am having the same issue but with other env-variables that I use in codegen tasks prior to running `next build`

**Reply by ehaynes99** [reply]
Yeah, it seems like it applies to all env vars. I get errors about about failing to substitute in my `.npmrc`
```
//npm.pkg.github.com/:_authToken=${GITHUB_NPM_TOKEN}
```

**Comment by omawhite** [comment]
Did you take a look at [this](https://turbo.build/blog/turbo-2-0#breaking-changes)? I found it a little more helpful than trying to look at the releases here on github for breaking changes.
